pingcap/tidb
# Makefile: make 'make check' faster

### What problem does this PR solve? 

Try to accelerate the unit test. 
`make check` is painfully slow, I make it faster in this commit. 

Before:

```
➜  tidb git:(master) ✗ time make check    
make check  44.87s user 34.64s system 86% cpu 1:31.63 total
```

After:
```
➜  tidb git:(make-check-accelerate) ✗ time make check
make check  6.87s user 1.77s system 113% cpu 7.607 total
```

### What is changed and how it works?

* remove vet check
* make lint check on ./...

#### 1.  vet is removed from `make check`, because from [Go1.10](https://golang.org/doc/go1.10#test) on, `vet` will run automatically when `go test` is called, and the [Go1.11](https://golang.org/doc/go1.11#test) also keeps this behavior.

To verify that, I tried this:

```
➜  session git:(master) ✗ cd session
➜  session git:(master) ✗ time go test -vet=off 1>/dev/null 2>/dev/null
go test -vet=off > /dev/null 2> /dev/null  7.26s user 1.28s system 63% cpu 13.403 total

➜  session git:(master) ✗ time go test 1>/dev/null 2>/dev/null   
go test > /dev/null 2> /dev/null  12.54s user 1.80s system 91% cpu 15.719 total
```

You can see that test with `-vet=off` is much faster.
Since each test package has already run `vet`, let `make check` run it is meaningless 

#### 2. `make lint` run this command:

> 	@CGO_ENABLED=0 retool do revive -formatter friendly -config revive.toml $(PACKAGES)

I find it downloads $(PACKAGES) from network, which is not a desired behavior.

To verify that, I tried to lint a non-local package (github.com/tiancaiamao/shen-go/runtime for example):

```
➜  tidb git:(master) ✗ retool do revive -formatter friendly -config revive.toml github.com/tiancaiamao/shen-go/runtime
go/build: importGo github.com/tiancaiamao/shen-go/runtime: exit status 1
go: finding github.com/tiancaiamao/shen-go/runtime latest
go: finding github.com/tiancaiamao/shen-go latest
go: extracting github.com/tiancaiamao/shen-go v0.0.0-20181124091523-b5e263ee27d0
-> unzip /media/genius/OS/project/pkg/mod/cache/download/github.com/tiancaiamao/shen-go/@v/v0.0.0-20181124091523-b5e263ee27d0.zip: malformed file path "ShenOSKernel-21.0/tests/depth'.shen": invalid char '\''
can't load package: package github.com/tiancaiamao/shen-go/runtime: unknown import path "github.com/tiancaiamao/shen-go/runtime": cannot find module providing package github.com/tiancaiamao/shen-go/runtime
```

#### 3. Another change is that I call `revive` directly instead of under the `retool do`.
Retool is introduced in https://github.com/pingcap/tidb/pull/7240, to specify tool version, now that we have Go module, it's possible to specify tool version in go.mod

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 - No code


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/8755)
